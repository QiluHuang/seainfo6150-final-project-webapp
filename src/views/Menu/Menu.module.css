.menu_row {
    display: grid;
    grid-template-columns: 15% 85%;
    width: 100%;
    /* flex-direction: row; */
    /* flex-wrap: wrap; */
}

.menu_sidebar {
    margin-top: 50px;
}



.menu_sidebar_ul {
    list-style-type: none;
    margin: 0;
    padding: 0px 10px 0px 10px;
    overflow: hidden;
    background-color: white;
    text-align-last: right;
    font-family: Monaco;
}

.menu_sidebar_ul a {
    text-decoration: none;
    color: black;
}

.menu_sidebar_li {
    font-size: 1em;
    padding: 0;
    margin: 0;
}

.menu_sidebar_li:hover:not(.active) {
    background-color:rgb(225, 225, 225);
}

.activeClassName  {
    color:green !important;
}


.menuList {
    width: 100%;
    /* font-family: 'Times New Roman', Times, serif; */
}


