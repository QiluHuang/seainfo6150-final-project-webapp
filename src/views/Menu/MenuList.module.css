.menuList_wrapper {
  /* text-align: center; */
  width: 100%;
}

.title {
  font-family: 'Times New Roman', Times, serif;
  font-size: 1.5em;
  text-align: center;
  text-decoration: underline;
}

.menuList_container {
  width: 100%;
  display: grid;
  grid-template-columns: repeat(4, 1fr);
  gap: 10px;
  font-family: Georgia, 'Times New Roman', Times, serif;
  /* list-style-type: none; */
  /* padding: 10px; */
  /* margin: 10px 10px 10px 10px; */
  
}


@media (max-width: 768px) {
  .menuList_container {
    grid-template-columns: repeat(2, 1fr);
  }
}

@media (min-width: 768px) and (max-width: 1390px) {
  .menuList_container {
    grid-template-columns: repeat(3, 1fr);
  }
}

