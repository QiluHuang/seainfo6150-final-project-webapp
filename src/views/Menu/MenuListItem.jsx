import React, { useState } from "react";
import { Switch, Route, Link, useRouteMatch } from "react-router-dom";

import PropTypes from "prop-types";
import styles from "./MenuListItem.module.css";
// import ArticleTextToggleButton from "../ArticleTextToggleButton/ArticleTextToggleButton.jsx";
import DishImageMedium from "../../components/Images/DishImageMedium/DishImageMedium.jsx";

const DishListItem = (props) => {
  return (
      <div className={styles.container}>
        <Link to={`/menu/${props.dish.categoryID}`}>
          <div className={styles.wrapper}>
            <div className = {styles.upper_image}>
              <DishImageMedium url={props.dish.image._url} title={props.dish.name}/>
            </div>
            <h5 className={styles.name}>{props.dish.category}</h5>
          </div>
        </Link>
      </div>
  );
};

DishListItem.propTypes = {
  dish: PropTypes.object.isRequired,
};
export default DishListItem;