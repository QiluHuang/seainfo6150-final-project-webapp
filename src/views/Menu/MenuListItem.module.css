.container {
    width: auto;
    display: flex;
    flex-flow: column wrap;
    justify-content: space-between;
    height: 100%;
    width: 100%;
    /* font-family: Arial */
  }
  
.container a {
  text-decoration: none;
}

.container:hover {
  box-shadow: 0 5px 10px 5px rgba(0, 0, 0, 0.1);
}

  
.wrapper {
  flex: 1;
  padding-left: 0px;
  text-align: center;
}
  
  
.upper_image {
    text-align: center;
    padding: 10px;
    /* padding-top: 10px;
    padding-bottom: 10px; */
}


.name {
  font-size: 1em;
  text-align: center;
}



