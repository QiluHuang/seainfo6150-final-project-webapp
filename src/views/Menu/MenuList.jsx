import React from 'react';
import { Switch, Route, Link, useRouteMatch } from "react-router-dom";
import PropTypes from "prop-types";
import styles from "./MenuList.module.css";

import MenuListItem from './MenuListItem.jsx';



const MenuList = (props) => {
    console.log(props);
    return ( 
        
        <div className={styles.menuList_wrapper}>
            <div className={styles.title}>
                <h3> MENU </h3>
            </div>
            <div className={styles.menuList_container}>
                {props.menuList.map((dish) =>  (
                    <MenuListItem dish={dish} key={dish.slug} />
                ))}
            </div>
        </div>
    );
}

MenuList.propTypes = {
    menuList: PropTypes.array.isRequired
}

export default MenuList;