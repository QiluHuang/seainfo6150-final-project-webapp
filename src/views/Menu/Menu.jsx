import React, { useEffect, useState } from 'react';
import { Switch, Route, Link, NavLink, useRouteMatch, Redirect } from "react-router-dom";
import styles from './Menu.module.css';

import MenuList from './MenuList.jsx';
import DishDetail from '../DishDetail/DishDetail.jsx';
import CategoryList from '../Category/CategoryList/CategoryList';



const Menu = () => {
    const [fetchedData, setFetchedData] = useState([]);
    const [fetchedDishData, setFetchedDishData] = useState([]);


    useEffect(() => {
      const fetchData = async () => {
        // performs a GET request
        const response = await fetch("https://demo1273150.mockable.io/menu");
        const responseJson = await response.json();
        setFetchedData(Object.values(responseJson));
      };
      const fetchDishData = async () => {
        // performs a GET request
        const response = await fetch("https://demo1273150.mockable.io/dishes");
        const responseJson = await response.json();
        setFetchedDishData(Object.values(responseJson));
      };
  
      if (!fetchedData.length) {
        fetchData();
      }
      if (!fetchedDishData.length) {
        fetchDishData();
      }
    }, [fetchedData, fetchedDishData]);
  
    let displayContent;
  
    if (fetchedData.length && fetchedDishData.length) {
    //   console.log(typeof fetchedData);
    //   console.log(fetchedData);
    //   console.log(fetchedDishData);


      displayContent = (
        <div className={styles.menu_row}>
            
            <div className={styles.menu_sidebar}>
                
                <ul className={styles.menu_sidebar_ul}>
                    <li className={styles.menu_sidebar_li}>                        
                        <NavLink to="/menu/content" activeClassName={styles.activeClassName}><h4>Content</h4></NavLink>
                    </li>
                    <li className={styles.menu_sidebar_li}>
                        <NavLink to={`/menu/${fetchedData[0].categoryID}`} activeClassName={styles.activeClassName}>
                            <div><h4>{fetchedData[0].category}</h4></div>
                        </NavLink>
                    </li>
                    <li className={styles.menu_sidebar_li}>
                        <NavLink to={`/menu/${fetchedData[1].categoryID}`} activeClassName={styles.activeClassName}>
                            <div><h4>{fetchedData[1].category}</h4></div>
                        </NavLink>
                    </li>
                    <li className={styles.menu_sidebar_li}>
                        <NavLink to={`/menu/${fetchedData[2].categoryID}`} activeClassName={styles.activeClassName}>
                            <div><h4>{fetchedData[2].category}</h4></div>
                        </NavLink>
                    </li>
                    <li className={styles.menu_sidebar_li}>
                        <NavLink to={`/menu/${fetchedData[3].categoryID}`} activeClassName={styles.activeClassName}>
                            <div><h4>{fetchedData[3].category}</h4></div>
                        </NavLink>
                    </li>
                    <li className={styles.menu_sidebar_li}>
                        <NavLink to={`/menu/${fetchedData[4].categoryID}`} activeClassName={styles.activeClassName}>
                            <div><h4>{fetchedData[4].category}</h4></div>
                        </NavLink>
                    </li>
                    <li className={styles.menu_sidebar_li}>
                        <NavLink to={`/menu/${fetchedData[5].categoryID}`} activeClassName={styles.activeClassName}>
                            <div><h4>{fetchedData[5].category}</h4></div>
                        </NavLink>
                    </li>
                    <li className={styles.menu_sidebar_li}>
                        <NavLink to={`/menu/${fetchedData[6].categoryID}`} activeClassName={styles.activeClassName}>
                            <div><h4>{fetchedData[6].category}</h4></div>
                        </NavLink>
                    </li>
                    <li className={styles.menu_sidebar_li} >
                        <NavLink to={`/menu/${fetchedData[7].categoryID}`} activeClassName={styles.activeClassName}>
                            <div><h4>{fetchedData[7].category}</h4></div>
                        </NavLink>
                    </li>
                </ul>
            </div>
            <div className={styles.menuList}>
                <Switch>
                    <Route 
                        path="/menu/content" 
                        exact
                        render={() => <MenuList menuList={fetchedData}/>}/>

                    <Route 
                        path={`/menu/:categoryID`} 
                        exact 
                        render={({match})=>(
                            <CategoryList categoryID={match.params.categoryID} 
                                          dishes={fetchedDishData}/> 
                        )}
                    />

                    <Route 
                        path={`/menu/:categoryID/:slug`}
                        exact
                        render={({match}) => (
                            <DishDetail slug={match.params.slug}
                                        dishes={fetchedDishData}
                            />)}
                    />

                    
                    <Redirect to="/menu/content" />
                </Switch>
            </div>

        </div>
      );
    } else {
      displayContent = <div>You have no data!</div>;
    }

 
    

    return displayContent;
}

export default Menu;