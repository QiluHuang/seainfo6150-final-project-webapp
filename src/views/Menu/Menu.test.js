import React from "react";
import { render } from "@testing-library/react";
import Menu from "./Menu.jsx";

describe("Menu tests", () => {
    it("renders the menu component correctly", () => {
    const { container } = render(<Menu />);
    expect(container).toMatchSnapshot();
    });
});