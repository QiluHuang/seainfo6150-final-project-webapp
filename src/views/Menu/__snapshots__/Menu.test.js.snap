// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu tests renders the menu component correctly 1`] = `
<div>
  <div>
    You have no data!
  </div>
</div>
`;
